From: Joerg Roedel <jroedel@suse.de>
Date: Fri, 19 Oct 2018 17:53:29 +0200
Subject: [PATCH] kvm/x86: kABI fix for vm_alloc/vm_free changes
Patch-mainline: no, SUSE-specific
References: bsc#1111506

Hide the new call-backs in x86-ops at the end of the struct
to not change the layout.

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/include/asm/kvm_host.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -941,8 +941,6 @@ struct kvm_x86_ops {
 	bool (*cpu_has_high_real_mode_segbase)(void);
 	void (*cpuid_update)(struct kvm_vcpu *vcpu);
 
-	struct kvm *(*vm_alloc)(void);
-	void (*vm_free)(struct kvm *);
 	int (*vm_init)(struct kvm *kvm);
 	void (*vm_destroy)(struct kvm *kvm);
 
@@ -1109,6 +1107,8 @@ struct kvm_x86_ops {
 
 #ifndef __GENKSYMS__
 	bool (*has_emulated_msr)(int index);
+	struct kvm *(*vm_alloc)(void);
+	void (*vm_free)(struct kvm *);
 #endif
 
 	bool (*need_emulation_on_page_fault)(struct kvm_vcpu *vcpu);
